client/state/happychat/selectors.js

```diff
@@ -97,16 +112,25 @@ export const getHappychatTimeline = createSelector(
 	state => map( state.happychat.timeline, 'id' )
 );
 
-export const canUserSendMessages = createSelector(
-	state => (
-		getHappychatConnectionStatus( state ) === 'connected' &&
-		! includes(
-			[	HAPPYCHAT_CHAT_STATUS_PENDING, HAPPYCHAT_CHAT_STATUS_MISSED,
-				HAPPYCHAT_CHAT_STATUS_ABANDONED ],
-			getHappychatChatStatus( state )
-		) && isHappychatAcceptingChats( state )
-	),
-	[ getHappychatConnectionStatus, getHappychatChatStatus ]
+/**
+ * Returns true if the user should be able to send messages to operators based on
+ * chat status. For example new chats and ongoing chats should be able to send messages,
+ * but blocked or pending chats should not.
+ * @param {Object} state - global redux state
+ * @return {Boolean} Whether the user is able to send messages
+ */
+export const canUserSendMessages = state => (
+	isHappychatAvailable( state ) &&
```

Comment:
Connection status is implicitly included as part of the `isHappychatAvailable` selector, which makes the same check on `isHappychatClientConnected` that was here before.

We _could_ explicitly include an `isHappychatClientConnected` check here, but I don't think we should — we should trust that `isHappychatAvailable` means "Happychat is able to take chats" regardless of connection state (for example, maybe in the future we allow users to write messages before a connection has been established — Happychat could still be considered "available" even without an active connection).